File: src/com/owncloud/android/lib/resources/notifications/RichObject.java
/*  Nextcloud Android Library is available under MIT license
 *   Copyright (C) 2017 Andy Scherzinger
 *   Copyright (C) 2017 Nextcloud GmbH
 *
 *   @author Andy Scherzinger
 *
 *   Permission is hereby granted, free of charge, to any person obtaining a copy
 *   of this software and associated documentation files (the "Software"), to deal
 *   in the Software without restriction, including without limitation the rights
 *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 *   copies of the Software, and to permit persons to whom the Software is
 *   furnished to do so, subject to the following conditions:
 *
 *   The above copyright notice and this permission notice shall be included in
 *   all copies or substantial portions of the Software.
 *
 *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
 *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
 *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
 *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
 *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
 *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 *   THE SOFTWARE.
 *
 */

package com.owncloud.android.lib.resources.notifications;

/**
 * Rich object data model providing rich data to be used within rich content, e.g.
 * in {@link Notification}s.
 */
public class RichObject {
    /**
     * A unique identifier for the object type.
     */
    public String type;

    /**
     * A short identifier of the object on the server (int or string).
     */
    public String id;

    /**
     * A name which should be used in the visual representation.
     */
    public String name;

    public RichObject() {

    }

    public RichObject(String type, String id, String name) {
        this.type = type;
        this.id = id;
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: library/src/main/java/com/owncloud/android/lib/resources/activities/model/RichObject.kt
/*
 * Nextcloud Android Library
 *
 * SPDX-FileCopyrightText: 2017-2024 Nextcloud GmbH and Nextcloud contributors
 * SPDX-FileCopyrightText: 2022 Álvaro Brey <alvaro@alvarobrey.com>
 * SPDX-FileCopyrightText: 2017 Alejandro Bautista <aleister09@gmail.com>
 * SPDX-License-Identifier: MIT
 */
package com.owncloud.android.lib.resources.activities.model

/**
 * RichObject Data Model
 */
class RichObject() {
    var type: String? = null
    var id: String? = null
    var name: String? = null
    var path: String? = null
    var link: String? = null
    var tag: String? = null

    @Suppress("LongParameterList")
    constructor(type: String?, id: String?, name: String?, path: String?, link: String?, tag: String?) : this() {
        this.type = type
        this.id = id
        this.name = name
        this.path = path
        this.link = link
        this.tag = tag
    }
}
